src/axom/inlet/tests/inlet_object.cpp

```diff
@@ -147,6 +147,82 @@ TYPED_TEST(inlet_object, simple_array_of_struct_verify_reqd)
   EXPECT_FALSE(inlet.verify());
 }
 
+TYPED_TEST(inlet_object, simple_array_of_struct_verify_empty)
+{
+  std::string testString = "foo = { }";
+  DataStore ds;
+  Inlet inlet = createBasicInlet<TypeParam>(&ds, testString);
+
+  auto& arr_table = inlet.addGenericArray("foo");
+  // Even though these are required, the input file should still
+  // "verify" as the array is empty
+  arr_table.addBool("bar", "bar's description").required(true);
+  arr_table.addBool("baz", "baz's description").required(true);
+
+  EXPECT_TRUE(inlet.verify());
+}
+
+TYPED_TEST(inlet_object, simple_array_of_struct_verify_lambda)
+{
+  std::string testString =
+    "foo = { [4] = { bar = true;}, "
+    "        [7] = { bar = false;} }";
+  DataStore ds;
+  Inlet inlet = createBasicInlet<TypeParam>(&ds, testString);
+
+  auto& arr_table = inlet.addGenericArray("foo");
+
+  arr_table.addBool("bar", "bar's description").required();
```

Comment:
Just checking -- the following two lines are equivalent to each other, right:
```cpp
   arr_table.addBool("bar", "bar's description").required(); 
   arr_table.addBool("bar", "bar's description").required(true); 
```

and similarly, the following two lines are equivalent to each other:
```cpp
   arr_table.addBool("bar", "bar's description"); 
   arr_table.addBool("bar", "bar's description").required(false); 
```

Reply:
Yes - in some cases, like https://github.com/LLNL/axom/blob/ec05a4704cc5b3c15043655cf1b5d49a591d1f9f/src/axom/inlet/tests/inlet_object.cpp#L128-L129

I've explicitly specified it one way or another with the intention of improving readability.